### ISSUE TITLE
Opening cache...newsboat: src/cache.cpp:1166: newsboat::SchemaVersion newsboat::Cache::get_schema_version(): Assertion `sqlite3_step(stmt) == SQLITE_DONE' failed. [1]    6329 abort (core dumped)  newsboat

### ISSUE THREAD

**Issue by jarbus** [newsboat/newsboat#1446]
**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**:
```
newsboat 2.22.1
System: Linux 5.4.91-1-lts (x86_64)
Compiler: g++ 10.2.0
ncurses: ncurses 6.2.20200212 (compiled with 6.2)
libcurl: libcurl/7.74.0 OpenSSL/1.1.1i zlib/1.2.11 zstd/1.4.8 libidn2/2.3.0 libpsl/0.21.1 (+libidn2/2.3.0) libssh2/1.9.0 nghttp2/1.41.0 (compiled with 7.74.0)
SQLite: 3.34.0 (compiled with 3.34.0)
libxml2: compiled with 2.9.10
```
Config file (copy from ~/.newsboat/config or ~/.config/newsboat/config):
```
# general settings
auto-reload no
max-items 50
bind-key SPACE macro-prefix
#mark-as-read-on-hover yes
toggleitemread-jumps-to-next-unread no
prepopulate-query-feeds yes

# reminder: N toggles read status

# external browser
external-url-viewer "urlview"
bind-key U show-urls
browser "linkhandler %u"
unbind-key v
macro v set browser "/usr/bin/mpv %u"; open-in-browser ; set browser "linkhandler %u"
macro a set browser "/usr/bin/mpv %u --no-video"; open-in-browser ; set browser "linkhandler %u" 
```

Steps to reproduce the issue:

1. `$ newsboat`

**Other info you think is relevant**:
Up until today newsboat worked fine, now instead of starting it aborts on the following failed assertion:

```
Opening cache...newsboat: src/cache.cpp:1166: newsboat::SchemaVersion newsboat::Cache::get_schema_version(): Assertion `sqlite3_step(stmt) == SQLITE_DONE' failed.
[1]    6329 abort (core dumped)  newsboat
```

The database is not corrupted, as I can open it in `sqlitebrowser` and view entries. What might the problem be? Is there something wrong with my version of SQL?

**Comment by Minoru** [comment]
What does `SELECT * FROM metadata` return? Or, in other words, what does the `metadata` table contain?

I doubt this has anything to do with SQLite. I'm using the same version, and I haven't seen any problems yet. And I'm pretty sure SQLite is more reliable than Newsboat :)

**Comment by jarbus** [comment]
Sorry for the late response, and thank you for your help! The table contains the following:

![image](https://user-images.githubusercontent.com/42819002/105937213-15280c00-6023-11eb-9b16-c151ef4c805b.png)

**Comment by Minoru** [comment]
Ah, just as I feared. There should be only one row in that table, and [I can't see how Newsboat could create two](https://github.com/newsboat/newsboat/blob/213be352e328d4fe5df834fc99798c2ed4c686fd/src/cache.cpp#L375-L390) :( Do you happen to remember what happened to cache.db before this problem appeared? You probably upgraded to 2.22.1 and the crash happened immediately? Was there any weirdness around that time, like power outages? Maybe Newsboat crashed for a different reason the first time, and when you re-ran it, it started crashing because of the assertion? I'd like to understand what led to this wrong data in the DB.

Meanwhile, do the following:
1. backup your cache.db :)
2. remove any of the two rows in the `metadata` table;
3. check that Newsboat starts okay now. Maybe keep a backup for a few days until you're sure that nothing else is broken.

**Comment by jarbus** [comment]
Wow, that fixed it! Thanks a bunch. Unsure of what might have happened. I sometimes will accidentally close a terminal that has newsboat open, so I'll need to delete the cache.db-lock file to get it to open again, but it's never caused any issues until now.

**Comment by Minoru** [comment]
> so I'll need to delete the cache.db-lock file to get it to open again

Wait, you mean *it won't start unless you delete the lock file*? This means the previous instance of Newsboat is not dead yet. (Or, with a likelihood of 2^-16, that another process was started and got the same process ID that's written in the lock file). Newsboat is not designed for multiple processes to access its DB simultaneously, so I wouldn't be surprised that *this* is the source of corruption.

And I don't see how Newsboat could survive the closing of its terminal either, unless you're running it under Tmux, Screen, or some other such thing. If you find that it *does* survive, and you can reproduce that consistently, please open a separate issue and we'll investigate that further. (FWIW, I just closed a terminal window with my copy of Newsboat, and it crashed as expected.)

Anyway, *this* issue is now resolved, so I'm closing it. Cheers!

**Comment by jarbus** [comment]
Interesting, I just tried closing my terminal with Newsboat open, and was able to re-launch Newsboat in a separate window without issue. I've experienced issues in the past where Newsboat wouldn't open because the database was locked, even though there was no other Newsboat window open, so I assumed I must have closed it incorrectly. The only solution I found was deleting the lock file, which would get Newsboat to open again.
